app/controllers/assessment/gradings_controller.rb

```diff
@@ -240,6 +243,16 @@ def auto_grading_keyword_grade(question, answer)
     [score, question.max_grade].min
   end
 
+  # Given a keyword, returns a sanitised regex which matches it. Keywords are assumed
+  # to be single alphanumeric words.
+  # This is used in auto-grading and ensures that the same matching strategy is used
+  # everywhere keyword matches are required.
+  def keyword_regex(keyword)
+    keyword = keyword.gsub(/[^\d\w]/, '')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.